### PR TITLE
309 blog template generator

### LINKED ISSUE #309: Feature: add blog template to application generator
### Which scope/s are relevant/related to the feature request?

create-analog

### Information

When generating a new project with `create-analog`, there should be an option to scaffold out a blog template application with support for content routes (and maybe Tailwind) included.

Includes
- / for home route that redirects to /blog
- /blog parent route
- /blog index route that lists posts
- /blog/[slug].page.ts that displays the blog post based on id
- Stub markdown file in the `/src/content` folder
- Minimal styles
- Prerendered route included in the `vite.config.ts`

### Describe any alternatives/workarounds you're currently using

_No response_

### I would be willing to submit a PR to fix this issue

- [ ] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by BaronVonPerko**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/analogjs/analog/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Which package are you modifying?

- [ ] vite-plugin-angular
- [ ] vite-plugin-nitro
- [ ] astro-angular
- [x] create-analog
- [ ] router
- [ ] platform
- [ ] content
- [ ] nx-plugin
- [ ] trpc

## What is the current behavior?

Adds a new option to build a new analog site with Angular 17 and a basic blog template.

Closes #309 

## What is the new behavior?

Generates a new site.  The home page redirects to `/blog`, where a list of content is displayed.  Clicking one of the content links takes the user to view the content of that blog post.

The styling is left very minimal to allow the developer to design their site how they'd like to.

image

image

image

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

I'd love to get feedback on this and if there is anything else that should be added!

### CODE REVIEW

File: packages/create-analog/template-angular-v17-blog/angular.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+              "buildTarget": "my-app:build:development",
+              "hmr": true
+            },
+            "production": {
+              "buildTarget": "my-app:build:production"
+            }
+          }
+        },
+        "test": {
+          "builder": "@nx/vite:test"
```
Comment (brandonroberts) [inline packages/create-analog/template-angular-v17-blog/angular.json:49]:
```suggestion
          "builder": "@analogjs/platform:vitest"
```

File: packages/create-analog/template-angular-v17-blog/angular.json [hunk 2]
```diff
@@ -0,0 +1,54 @@
+{
+  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
+  "version": 1,
+  "newProjectRoot": "projects",
+  "projects": {
+    "my-app": {
```
Comment (brandonroberts) [inline packages/create-analog/template-angular-v17-blog/angular.json:6]:
```suggestion
    "blog": {
```

File: packages/create-analog/template-angular-v17-blog/angular.json [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      "projectType": "application",
+      "root": "",
+      "sourceRoot": "src",
+      "prefix": "app",
```
Comment (brandonroberts) [inline packages/create-analog/template-angular-v17-blog/angular.json:10]:
```suggestion
      "prefix": "blog",
```

File: packages/create-analog/template-angular-v17-blog/angular.json [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+              "sourcemap": false,
+              "mode": "production"
+            }
+          }
+        },
+        "serve": {
+          "builder": "@analogjs/platform:vite-dev-server",
+          "defaultConfiguration": "development",
+          "options": {
+            "buildTarget": "my-app:build",
```
Comment (brandonroberts) [inline packages/create-analog/template-angular-v17-blog/angular.json:35]:
```suggestion
            "buildTarget": "blog:build",
```

File: packages/create-analog/template-angular-v17-blog/angular.json [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+            "port": 5173
+          },
+          "configurations": {
+            "development": {
+              "buildTarget": "my-app:build:development",
```
Comment (brandonroberts) [inline packages/create-analog/template-angular-v17-blog/angular.json:40]:
```suggestion
              "buildTarget": "blog:build:development",
```

File: packages/create-analog/template-angular-v17-blog/angular.json [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+              "hmr": true
+            },
+            "production": {
+              "buildTarget": "my-app:build:production"
```
Comment (brandonroberts) [inline packages/create-analog/template-angular-v17-blog/angular.json:44]:
```suggestion
              "buildTarget": "blog:build:production"
```

File: packages/create-analog/template-angular-v17-blog/index.html [hunk 7]
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <title>MyApp</title>
```
Comment (brandonroberts) [inline packages/create-analog/template-angular-v17-blog/index.html:5]:
```suggestion
    <title>My Blog</title>
```

File: packages/create-analog/template-angular-v17-blog/package.json [hunk 8]
```diff
@@ -0,0 +1,51 @@
+{
+  "name": "my-app",
```
Comment (brandonroberts) [inline packages/create-analog/template-angular-v17-blog/package.json:2]:
```suggestion
  "name": "blog",
```

File: packages/create-analog/template-angular-v17-blog/src/app/pages/blog/index.page.ts [hunk 9]
```diff
@@ -0,0 +1,34 @@
+import { Component } from '@angular/core';
+import { injectContentFiles } from '@analogjs/content';
+import { NgForOf } from '@angular/common';
+import PostAttributes from '../../post-attributes';
+
+@Component({
+  selector: 'app-blog',
+  standalone: true,
+  imports: [NgForOf],
```
Comment (brandonroberts) [inline packages/create-analog/template-angular-v17-blog/src/app/pages/blog/index.page.ts:9]: New Control Flow here?

File: packages/create-analog/index.js [hunk 10]
```diff
@@ -24,6 +24,11 @@ const APPS = [
         display: 'TypeScript',
         color: green,
       },
+      {
+        name: 'angular-v17-blog',
```
Comment (brandonroberts) [inline packages/create-analog/index.js:28]: I'd just use `blog` here and assume you're using the latest version of Angular and Analog. And change the folder name to `template-blog`.

```suggestion
        name: 'blog',
```

File: packages/create-analog/template-angular-v17-blog/_gitignore [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+.sass-cache/
+/connect.lock
+/coverage
+/libpeerconnection.log
+testem.log
+/typings
+
+# System files
+.DS_Store
+Thumbs.db
```
Comment (brandonroberts) [inline packages/create-analog/template-angular-v17-blog/_gitignore:42]: Let's add `/.nx/cache` to this file also

File: packages/create-analog/template-angular-v17-blog/src/app/pages/blog/[slug].page.ts [hunk 12]
```diff
@@ -0,0 +1,26 @@
+import { Component } from '@angular/core';
+import { injectContent, MarkdownComponent } from '@analogjs/content';
+import { AsyncPipe, NgIf } from '@angular/common';
+import PostAttributes from '../../post-attributes';
```
Comment (brandonroberts) [inline packages/create-analog/template-angular-v17-blog/src/app/pages/blog/[slug].page.ts:4]: nit: add extra line

```suggestion

import PostAttributes from '../../post-attributes';
```

File: packages/create-analog/template-angular-v17-blog/src/app/pages/blog/[slug].page.ts [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+
+@Component({
+  selector: 'app-blog-post',
+  standalone: true,
+  imports: [AsyncPipe, NgIf, MarkdownComponent],
```
Comment (brandonroberts) [inline packages/create-analog/template-angular-v17-blog/src/app/pages/blog/[slug].page.ts:9]: New Control flow here?

**Review by brandonroberts — CHANGES_REQUESTED** [review]
Thanks! Some relatively minor changes

File: packages/create-analog/template-angular-v17-blog/src/app/pages/blog/index.page.ts (hunk 9 above)
Comment (BaronVonPerko) [reply packages/create-analog/template-angular-v17-blog/src/app/pages/blog/index.page.ts:9]: I was thinking about that... are we ok using dev preview features? 🤔

Comment (brandonroberts) [reply packages/create-analog/template-angular-v17-blog/src/app/pages/blog/index.page.ts:9]: Good point. I'm good with using it for a blog template. We can always change the template if its confusing

Comment (BaronVonPerko) [reply packages/create-analog/template-angular-v17-blog/src/app/pages/blog/index.page.ts]: Awesome :) just pushed that change as well as renamed the template directroy to `template-blog`.

File: packages/create-analog/template-blog/src/app/pages/blog/index.page.ts [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+import { injectContentFiles } from '@analogjs/content';
+import PostAttributes from '../../post-attributes';
+
+@Component({
+  selector: 'app-blog',
+  standalone: true,
+  template: `
+    <h1>Blog Archive</h1>
+    @for (post of posts; track post.attributes.slug) {
+    <a [href]="'/blog/' + post.attributes.slug">
```
Comment (brandonroberts) [inline packages/create-analog/template-blog/src/app/pages/blog/index.page.ts:11]:
```suggestion
    <a [routerLink]="['/blog/', post.attributes.slug]">
```

File: packages/create-analog/template-blog/src/app/pages/blog/index.page.ts [hunk 15]
```diff
@@ -0,0 +1,34 @@
+import { Component } from '@angular/core';
+import { injectContentFiles } from '@analogjs/content';
+import PostAttributes from '../../post-attributes';
+
+@Component({
```
Comment (brandonroberts) [inline packages/create-analog/template-blog/src/app/pages/blog/index.page.ts:5]: This component should have a `RouterLink` in the `imports`

File: packages/create-analog/template-blog/src/app/app.config.ts [hunk 16]
```diff
@@ -0,0 +1,14 @@
+import { provideHttpClient } from '@angular/common/http';
+import { ApplicationConfig } from '@angular/core';
+import { provideClientHydration } from '@angular/platform-browser';
+import { provideFileRouter } from '@analogjs/router';
+import { provideContent, withMarkdownRenderer } from '@analogjs/content';
+
+export const appConfig: ApplicationConfig = {
+  providers: [
+    provideFileRouter(),
+    provideHttpClient(),
```
Comment (brandonroberts) [inline packages/create-analog/template-blog/src/app/app.config.ts:10]: Let's default to withFetch also to get rid of the console warning.

```suggestion
    provideHttpClient(withFetch()),
```

Comment (BaronVonPerko) [reply packages/create-analog/template-blog/src/app/app.config.ts]: There's something new I learned 😄

File: packages/create-analog/template-blog/src/app/pages/blog/index.page.ts (hunk 14 above)
Comment (BaronVonPerko) [reply packages/create-analog/template-blog/src/app/pages/blog/index.page.ts]: Pushed a change to resolve this (as well as the next comment)

**Review by brandonroberts — APPROVED** [review]

**Comment by brandonroberts** [comment]
Thanks @BaronVonPerko!

**Pull request merged**